### ISSUE TITLE
Support for materialized views

### ISSUE THREAD

**Issue by SonOfLilit** [edgedb/edgedb#3432]
https://www.postgresql.org/docs/9.3/rules-materializedviews.html

This is a classic RDBMS feature that you tend to discover you need only a few years into a successful project, after you hire a DBA and they explain to you that it exists and your analytics queries can be optimized with it.

It's important to me that such features exist (or are planned to be supported soon) if I'm about to choose an RDBMS for a project that might grow.

**Comment by jnak** [comment]
Yes, it would be amazing to have materialized computed properties. I could see the feature being used extensively if EdgeDB nailed it and made it dead simple to materialize those.

**Comment by choibles** [comment]
Just want to echo my support for this feature. Has there been any further consideration of its priority in the EdgeDB roadmap?

Believe there's a strong use case for materialized views in increasing the queryability of types based on their computed properties. The present alternative seems to be defining real (not computed) properties on relevant types and attempting to use triggers to keep them updated to the canonical value (which gets messy quite fast, especially if there's inheritance involved). Or writing to a separate type or even db..
